diff --git a/pkg/resource/client.go b/pkg/resource/client.go
index cd52c30..6e631f9 100644
--- a/pkg/resource/client.go
+++ b/pkg/resource/client.go
@@ -31,6 +31,7 @@ func (clientConfigFn ClientConfigFunc) clientForGroupVersion(gv schema.GroupVers
+	cfg = rest.CopyConfig(cfg)
 	if negotiatedSerializer != nil {
 		cfg.ContentConfig.NegotiatedSerializer = negotiatedSerializer
 	}
 	cfg.GroupVersion = &gv
 	if len(gv.Group) == 0 {
 		cfg.APIPath = "/api"
@@ -46,6 +47,7 @@ func (clientConfigFn ClientConfigFunc) unstructuredClientForGroupVersion(gv sche
 	if err != nil {
 		return nil, err
 	}
+	cfg = rest.CopyConfig(cfg)
 	cfg.ContentConfig = UnstructuredPlusDefaultContentConfig()
 	cfg.GroupVersion = &gv
 	if len(gv.Group) == 0 {
