diff -ur -x '*~' -x target -x dar -x .daml daml/canton-coin/daml/CC/Coin.daml daml/canton-coin-upgrade/daml/CC/Coin.daml
--- daml/canton-coin/daml/CC/Coin.daml
+++ daml/canton-coin-upgrade/daml/CC/Coin.daml
@@ -36,6 +36,7 @@
     svc : Party
     owner : Party
     amount : ExpiringAmount
+    upgrade : Optional ()
   where
     signatory svc, owner
     ensure validExpiringAmount amount
diff -ur -x '*~' -x target -x dar -x .daml daml/canton-coin/daml/CC/CoinImport.daml daml/canton-coin-upgrade/daml/CC/CoinImport.daml
--- daml/canton-coin/daml/CC/CoinImport.daml
+++ daml/canton-coin-upgrade/daml/CC/CoinImport.daml
@@ -39,6 +39,7 @@
               svc = svc
               owner = receiver
               amount = coin.amount
+              upgrade = Some ()
             return $ Some CoinCreateSummary with
               coin = coinCid
               -- NOTE: this price is not the price at which the original coin was created.
diff -ur -x '*~' -x target -x dar -x .daml daml/canton-coin/daml/CC/CoinRules.daml daml/canton-coin-upgrade/daml/CC/CoinRules.daml
--- daml/canton-coin/daml/CC/CoinRules.daml
+++ daml/canton-coin-upgrade/daml/CC/CoinRules.daml
@@ -34,6 +34,7 @@
     svc : Party
     configSchedule : Schedule Time (CoinConfig USD)
     isDevNet : Bool
+    upgrade : Optional ()
   where
     ensure validSchedule configSchedule validCoinConfig
 
@@ -90,6 +91,7 @@
                             receiverFeeRatio = 1.0 -- ensure the provider pays only the traffic cost
                             lock = None
                         ]
+              upgrade = Some ()
             -- We don't issue any rewards as part of the transfer, as these are only for the burned fees.
             rewardsConfig = RewardsIssuanceConfig with
               issueAppRewards = False
@@ -171,6 +173,7 @@
         coin <- create Coin with
                   owner = receiver
                   amount = expiringAmount
+                  upgrade = Some ()
                   ..
         return CoinCreateSummary with
                   coin
@@ -554,6 +557,7 @@
   return TransferResult with
     round = csum.openRound.round
     summary = summary
+    upgrade = Some ()
     ..
 
 
@@ -857,11 +861,13 @@
             svc
             owner = sender
             amount = expiringAmount transferConfigCC.holdingFee summary.senderChangeAmount currentOpenRound.round
+            upgrade = Some ()
     createOutput out = do
       let coin = Coin with
             svc
             owner = out.owner
             amount = expiringAmount transferConfigCC.holdingFee out.amount currentOpenRound.round
+            upgrade = Some ()
 
       case out.lock of
         None -> TransferResultCoin <$> create coin
@@ -990,6 +996,7 @@
     provider : Party
     inputs : [TransferInput]
     outputs : [TransferOutput]
+    upgrade : Optional ()
   deriving (Eq, Ord, Show)
 
 -- | An individual output for a batch transfer.
@@ -1015,6 +1022,7 @@
     senderChangeCoin: Optional (ContractId Coin)
       -- ^ Optional reference to the coin for the change returned to the sender.
       -- Only created if there was some change to be returned after deducting the fee for returning change.
+    upgrade : Optional ()
   deriving (Show, Eq)
 
 data BuyMemberTrafficResult = BuyMemberTrafficResult with
diff -ur -x '*~' -x target -x dar -x .daml daml/canton-coin/daml.yaml daml/canton-coin-upgrade/daml.yaml
--- daml/canton-coin/daml.yaml
+++ daml/canton-coin-upgrade/daml.yaml
@@ -6,7 +6,7 @@
 sdk-version: 3.0.0-snapshot.20240209.12740.0.v499cb7f6
 name: canton-coin
 source: daml
-version: 0.1.0
+version: 0.2.0
 dependencies:
   - daml-prim
   - daml-stdlib
