diff -ur -x '*~' -x target -x dar -x .daml ../daml/canton-coin/daml/CC/Coin.daml ../daml/canton-coin-upgrade/daml/CC/Coin.daml
--- ../daml/canton-coin/daml/CC/Coin.daml
+++ ../daml/canton-coin-upgrade/daml/CC/Coin.daml
@@ -35,6 +35,7 @@
     svc : Party
     owner : Party
     amount : ExpiringAmount
+    upgrade : Optional ()
   where
     signatory svc, owner
     ensure validExpiringAmount amount
diff -ur -x '*~' -x target -x dar -x .daml ../daml/canton-coin/daml/CC/CoinImport.daml ../daml/canton-coin-upgrade/daml/CC/CoinImport.daml
--- ../daml/canton-coin/daml/CC/CoinImport.daml
+++ ../daml/canton-coin-upgrade/daml/CC/CoinImport.daml
@@ -39,6 +39,7 @@
               svc = svc
               owner = receiver
               amount = coin.amount
+              upgrade = Some ()
             return $ Some CoinCreateSummary with
               coin = coinCid
               -- NOTE: this price is not the price at which the original coin was created.
diff -ur -x '*~' -x target -x dar -x .daml ../daml/canton-coin/daml/CC/CoinRules.daml ../daml/canton-coin-upgrade/daml/CC/CoinRules.daml
--- ../daml/canton-coin/daml/CC/CoinRules.daml
+++ ../daml/canton-coin-upgrade/daml/CC/CoinRules.daml
@@ -32,6 +32,7 @@
     svc : Party
     configSchedule : Schedule Time (CoinConfig USD)
     isDevNet : Bool
+    upgrade : Optional ()
   where
     ensure validSchedule configSchedule validCoinConfig
 
@@ -86,6 +87,7 @@
                             receiverFeeRatio = 0.0
                             lock = None
                         ]
+              upgrade = Some ()
         result <- exercisePaymentTransfer context transfer
         let [TransferResultCoin coin] = result.createdCoins
         -- burn the coin created for the SVC
@@ -153,6 +155,7 @@
         coin <- create Coin with
                   owner = receiver
                   amount = expiringAmount
+                  upgrade = Some ()
                   ..
         return CoinCreateSummary with
                   coin
@@ -514,6 +517,7 @@
   return TransferResult with
     round = csum.openRound.round
     summary = summary
+    upgrade = Some ()
     ..
 
 data TransferContextSummary = TransferContextSummary with
@@ -778,11 +782,13 @@
             svc
             owner = sender
             amount = expiringAmount transferConfigCC.holdingFee summary.senderChangeAmount currentOpenRound.round
+            upgrade = Some ()
     createOutput out = do
       let coin = Coin with
             svc
             owner = out.owner
             amount = expiringAmount transferConfigCC.holdingFee out.amount currentOpenRound.round
+            upgrade = Some ()
 
       case out.lock of
         None -> TransferResultCoin <$> create coin
@@ -888,6 +894,7 @@
     provider : Party
     inputs : [TransferInput]
     outputs : [TransferOutput]
+    upgrade : Optional ()
   deriving (Eq, Ord, Show)
 
 -- | An individual output for a batch transfer.
@@ -913,6 +920,7 @@
     senderChangeCoin: Optional (ContractId Coin)
       -- ^ Optional reference to the coin for the change returned to the sender.
       -- Only created if there was some change to be returned after deducting the fee for returning change.
+    upgrade : Optional ()
   deriving (Show, Eq)
 
 -- | Summary of input and output amounts and fees paid.
diff -ur -x '*~' -x target -x dar -x .daml ../daml/canton-coin/daml/CC/Scripts/TestCoinImport.daml ../daml/canton-coin-upgrade/daml/CC/Scripts/TestCoinImport.daml
--- ../daml/canton-coin/daml/CC/Scripts/TestCoinImport.daml
+++ ../daml/canton-coin-upgrade/daml/CC/Scripts/TestCoinImport.daml
@@ -59,7 +59,7 @@
     openRoundCid = newOpenRoundCid
 
   -- the new coin has been created
-  let newCoin = Coin with svc = newApp.svc; owner = alice; amount = coin1.amount
+  let newCoin = Coin with svc = newApp.svc; owner = alice; amount = coin1.amount; upgrade = Some ()
   newCoins <- query @Coin newApp.svc
   map snd newCoins === [newCoin]
 
diff -ur -x '*~' -x target -x dar -x .daml ../daml/canton-coin/daml/CC/Scripts/TestCoinRulesTransfer.daml ../daml/canton-coin-upgrade/daml/CC/Scripts/TestCoinRulesTransfer.daml
--- ../daml/canton-coin/daml/CC/Scripts/TestCoinRulesTransfer.daml
+++ ../daml/canton-coin-upgrade/daml/CC/Scripts/TestCoinRulesTransfer.daml
@@ -69,6 +69,7 @@
                       holders = [alice.primaryParty, bob.primaryParty, bob.primaryParty] -- duplicates get removed
 
                 ]
+      upgrade = Some ()
     context = context.context
 
   let config = scaleFees 1.0 defaultTransferConfig
@@ -148,6 +149,7 @@
                       amount = 10.0
                       lock = None
                       receiverFeeRatio = 0.4 ]
+        upgrade = Some ()
       context = contextWithRound1.context
 
   -- Input amount is exactly 100.0.
@@ -227,6 +229,7 @@
                       amount
                       lock = None
                       receiverFeeRatio = 1.0]
+        upgrade = Some ()
       context = transferContext
 
 testSenderChangeAndOutput: Script ()
@@ -269,6 +272,7 @@
                     amount = outputAmount3
                     lock = None
                 ]
+      upgrade = Some ()
     context = context.context
   assertMsg "senderChangeAmount > 0"
     (transferResult.summary.senderChangeAmount > 0.0)
@@ -318,6 +322,7 @@
                     amount = outputAmount3
                     lock = None
                 ]
+      upgrade = Some ()
     context = context.context
   assertMsg "senderChangeAmount == 0"
     (transferResult.summary.senderChangeAmount == 0.0)
@@ -359,6 +364,7 @@
                     lock = None
 
                 ]
+      upgrade = Some ()
     context = context.context
   assertMsg "senderChangeAmount == 0"
     (transferResult.summary.senderChangeAmount == 0.0)
diff -ur -x '*~' -x target -x dar -x .daml ../daml/canton-coin/daml/CC/Scripts/TestCoinRulesTransferLimits.daml ../daml/canton-coin-upgrade/daml/CC/Scripts/TestCoinRulesTransferLimits.daml
--- ../daml/canton-coin/daml/CC/Scripts/TestCoinRulesTransferLimits.daml
+++ ../daml/canton-coin-upgrade/daml/CC/Scripts/TestCoinRulesTransferLimits.daml
@@ -44,6 +44,7 @@
           provider = alice.primaryParty
           inputs = inputCoins
           outputs = []
+          upgrade = Some ()
         context = context
         expectedError = "More than the maximum number of inputs"
     return ()
@@ -63,6 +64,7 @@
           provider = alice.primaryParty
           inputs = [InputCoin coin]
           outputs = outputCoins
+          upgrade = Some ()
         context = context
         expectedError = "More than the maximum number of outputs"
     return ()
@@ -94,6 +96,7 @@
                 expiresAt = now `addRelTime` (minutes 1)
               receiverFeeRatio = 0.0
           ]
+        upgrade = Some ()
       context = context
       expectedError = "Fewer than the allowed number of lock holders"
     return ()
diff -ur -x '*~' -x target -x dar -x .daml ../daml/canton-coin/daml/CC/Scripts/TestComputeFees.daml ../daml/canton-coin-upgrade/daml/CC/Scripts/TestComputeFees.daml
--- ../daml/canton-coin/daml/CC/Scripts/TestComputeFees.daml
+++ ../daml/canton-coin-upgrade/daml/CC/Scripts/TestComputeFees.daml
@@ -124,6 +124,7 @@
                       amount = 10.0
                       lock = None
                       receiverFeeRatio = 1.0]
+        upgrade = Some ()
       context = context.context
 
   -- Actual holding fee should be based on the coin price when the coin was created, not the current one
diff -ur -x '*~' -x target -x dar -x .daml ../daml/canton-coin/daml/CC/Scripts/TestDesignExample.daml ../daml/canton-coin-upgrade/daml/CC/Scripts/TestDesignExample.daml
--- ../daml/canton-coin/daml/CC/Scripts/TestDesignExample.daml
+++ ../daml/canton-coin-upgrade/daml/CC/Scripts/TestDesignExample.daml
@@ -58,6 +58,7 @@
                      lock = None
                      receiverFeeRatio = 1.0
                  ]
+               upgrade = Some ()
          exerciseAppTransfer transferContext transfer
 
 -- | This matches the example in the design doc
@@ -190,10 +191,12 @@
       svc = app.svc
       owner = alice.primaryParty
       amount = expiringAmount transferConfig.holdingFee 1000.25 c2.round
+      upgrade = Some ()
   change === Coin with
     svc = app.svc
     owner = alice.primaryParty
     amount = expiringAmount transferConfig.holdingFee senderChangeAmount c2.round
+    upgrade = Some ()
 
   payment === Coin with
     svc = app.svc
@@ -201,6 +204,7 @@
     amount =
       let amount = arbiterPaymentAmount - transferredCoinFee
       in expiringAmount transferConfig.holdingFee amount c2.round
+    upgrade = Some ()
 
   -- check app reward issued by the escrow refresh
   [(_, appRewardCoupon)] <- queryFilter @AppRewardCoupon app.svc (\reward -> reward.round.number == 50002)
diff -ur -x '*~' -x target -x dar -x .daml ../daml/canton-coin/daml/CC/Scripts/TestLockAndCoinExpiry.daml ../daml/canton-coin-upgrade/daml/CC/Scripts/TestLockAndCoinExpiry.daml
--- ../daml/canton-coin/daml/CC/Scripts/TestLockAndCoinExpiry.daml
+++ ../daml/canton-coin-upgrade/daml/CC/Scripts/TestLockAndCoinExpiry.daml
@@ -251,6 +251,7 @@
                 expiresAt = now `addRelTime` lockDuration
               receiverFeeRatio = 0.0
           ]
+        upgrade = Some ()
   (openRound, _) <- getLatestOpenRound app
   featuredAppRight <- getFeaturedAppRight app owner.primaryParty
 
diff -ur -x '*~' -x target -x dar -x .daml ../daml/canton-coin/daml/CC/Scripts/TestRewardComputation.daml ../daml/canton-coin-upgrade/daml/CC/Scripts/TestRewardComputation.daml
--- ../daml/canton-coin/daml/CC/Scripts/TestRewardComputation.daml
+++ ../daml/canton-coin-upgrade/daml/CC/Scripts/TestRewardComputation.daml
@@ -60,6 +60,7 @@
                     amount = 8.0
                     lock = None
                 ]
+      upgrade = Some ()
     context = context.context
 
   issuingMiningRounds <- queryFilter @IssuingMiningRound app.svc (\r -> r.round == Round 0)
diff -ur -x '*~' -x target -x dar -x .daml ../daml/canton-coin/daml/CC/Scripts/Util.daml ../daml/canton-coin-upgrade/daml/CC/Scripts/Util.daml
--- ../daml/canton-coin/daml/CC/Scripts/Util.daml
+++ ../daml/canton-coin-upgrade/daml/CC/Scripts/Util.daml
@@ -64,6 +64,7 @@
   _ <- submit svc $ createCmd CoinRules with
     configSchedule = defaultCoinConfigSchedule
     isDevNet = True
+    upgrade = Some ()
     ..
 
   submitExerciseCoinRulesByKey app [svc] [] CoinRules_Bootstrap_Rounds with
@@ -313,6 +314,7 @@
               receiverFeeRatio = transferFeeRatio
               lock = None
           ]
+        upgrade = Some ()
   (openRound, _) <- getLatestOpenRound app
   featuredAppRight <- getFeaturedAppRight app sender.primaryParty
   void $ submitExerciseCoinRulesByKey app [sender.primaryParty, recipient] readAs
@@ -341,6 +343,7 @@
                 expiresAt = now `addRelTime` lockDuration
               receiverFeeRatio = 0.0
           ]
+        upgrade = Some ()
   (openRound, _) <- getLatestOpenRound app
   featuredAppRight <- getFeaturedAppRight app sender.primaryParty
   void $ submitExerciseCoinRulesByKey app [sender.primaryParty, lockHolder] readAs
@@ -386,6 +389,7 @@
               map (InputValidatorRewardCoupon . fst) validatorRewardCoupons ++
               coins
           outputs = []
+          upgrade = Some ()
     when (not (null appRewardCoupons) || not (null validatorRewardCoupons) || length coins > 1) $ void $
       submitExerciseCoinRulesByKey app [user] readAs CoinRules_Transfer with
         transfer
diff -ur -x '*~' -x target -x dar -x .daml ../daml/canton-coin/daml.yaml ../daml/canton-coin-upgrade/daml.yaml
--- ../daml/canton-coin/daml.yaml
+++ ../daml/canton-coin-upgrade/daml.yaml
@@ -6,7 +6,7 @@
 sdk-version: 2.9.0-snapshot.20231211.12491.0.vbb56339f
 name: canton-coin
 source: daml
-version: 0.1.0
+version: 0.2.0
 dependencies:
   - daml-prim
   - daml-stdlib
