diff -ur -x '*~' -x target -x dar -x .daml ../daml/canton-coin/daml/CC/Coin.daml ../daml/canton-coin-upgrade/daml/CC/Coin.daml
--- ../daml/canton-coin/daml/CC/Coin.daml
+++ ../daml/canton-coin-upgrade/daml/CC/Coin.daml
@@ -35,6 +35,7 @@
     svc : Party
     owner : Party
     amount : ExpiringAmount
+    upgrade : Optional ()
   where
     signatory svc, owner
     ensure validExpiringAmount amount
diff -ur -x '*~' -x target -x dar -x .daml ../daml/canton-coin/daml/CC/CoinImport.daml ../daml/canton-coin-upgrade/daml/CC/CoinImport.daml
--- ../daml/canton-coin/daml/CC/CoinImport.daml
+++ ../daml/canton-coin-upgrade/daml/CC/CoinImport.daml
@@ -39,6 +39,7 @@
               svc = svc
               owner = receiver
               amount = coin.amount
+              upgrade = Some ()
             return $ Some CoinCreateSummary with
               coin = coinCid
               -- NOTE: this price is not the price at which the original coin was created.
diff -ur -x '*~' -x target -x dar -x .daml ../daml/canton-coin/daml/CC/CoinRules.daml ../daml/canton-coin-upgrade/daml/CC/CoinRules.daml
--- ../daml/canton-coin/daml/CC/CoinRules.daml
+++ ../daml/canton-coin-upgrade/daml/CC/CoinRules.daml
@@ -33,6 +33,7 @@
     svc : Party
     configSchedule : Schedule Time (CoinConfig USD)
     isDevNet : Bool
+    upgrade : Optional ()
   where
     ensure validSchedule configSchedule validCoinConfig
 
@@ -87,6 +88,7 @@
                             receiverFeeRatio = 0.0
                             lock = None
                         ]
+              upgrade = Some ()
         result <- exercisePaymentTransfer context transfer
         let [TransferResultCoin coin] = result.createdCoins
         -- burn the coin created for the SVC
@@ -154,6 +156,7 @@
         coin <- create Coin with
                   owner = receiver
                   amount = expiringAmount
+                  upgrade = Some ()
                   ..
         return CoinCreateSummary with
                   coin
@@ -531,6 +534,7 @@
   return TransferResult with
     round = csum.openRound.round
     summary = summary
+    upgrade = Some ()
     ..
 
 data TransferContextSummary = TransferContextSummary with
@@ -822,11 +826,13 @@
             svc
             owner = sender
             amount = expiringAmount transferConfigCC.holdingFee summary.senderChangeAmount currentOpenRound.round
+            upgrade = Some ()
     createOutput out = do
       let coin = Coin with
             svc
             owner = out.owner
             amount = expiringAmount transferConfigCC.holdingFee out.amount currentOpenRound.round
+            upgrade = Some ()
 
       case out.lock of
         None -> TransferResultCoin <$> create coin
@@ -933,6 +939,7 @@
     provider : Party
     inputs : [TransferInput]
     outputs : [TransferOutput]
+    upgrade : Optional ()
   deriving (Eq, Ord, Show)
 
 -- | An individual output for a batch transfer.
@@ -958,6 +965,7 @@
     senderChangeCoin: Optional (ContractId Coin)
       -- ^ Optional reference to the coin for the change returned to the sender.
       -- Only created if there was some change to be returned after deducting the fee for returning change.
+    upgrade : Optional ()
   deriving (Show, Eq)
 
 -- | Summary of input and output amounts and fees paid.
diff -ur -x '*~' -x target -x dar -x .daml ../daml/canton-coin/daml.yaml ../daml/canton-coin-upgrade/daml.yaml
--- ../daml/canton-coin/daml.yaml
+++ ../daml/canton-coin-upgrade/daml.yaml
@@ -6,7 +6,7 @@
 sdk-version: 2.9.0-snapshot.20231221.12524.0.vc998203f
 name: canton-coin
 source: daml
-version: 0.1.0
+version: 0.2.0
 dependencies:
   - daml-prim
   - daml-stdlib
