diff -ur -x '*~' -x target -x dar -x .daml daml/canton-coin-test/daml/CC/Scripts/TestCoinImport.daml daml/canton-coin-test-upgrade/daml/CC/Scripts/TestCoinImport.daml
--- daml/canton-coin-test/daml/CC/Scripts/TestCoinImport.daml
+++ daml/canton-coin-test-upgrade/daml/CC/Scripts/TestCoinImport.daml
@@ -60,7 +60,7 @@
     openRoundCid = newOpenRoundCid
 
   -- the new coin has been created
-  let newCoin = Coin with svc = newApp.svc; owner = alice; amount = coin1.amount
+  let newCoin = Coin with svc = newApp.svc; owner = alice; amount = coin1.amount; upgrade = Some ()
   newCoins <- query @Coin newApp.svc
   map snd newCoins === [newCoin]
 
diff -ur -x '*~' -x target -x dar -x .daml daml/canton-coin-test/daml/CC/Scripts/TestCoinRulesTransfer.daml daml/canton-coin-test-upgrade/daml/CC/Scripts/TestCoinRulesTransfer.daml
--- daml/canton-coin-test/daml/CC/Scripts/TestCoinRulesTransfer.daml
+++ daml/canton-coin-test-upgrade/daml/CC/Scripts/TestCoinRulesTransfer.daml
@@ -69,6 +69,7 @@
                       holders = [alice.primaryParty, bob.primaryParty, bob.primaryParty] -- duplicates get removed
 
                 ]
+      upgrade = Some ()
     context = context.context
 
   let config = scaleFees 1.0 defaultTransferConfig
@@ -150,6 +151,7 @@
                       amount = 10.0
                       lock = None
                       receiverFeeRatio = 0.4 ]
+        upgrade = Some ()
       context = contextWithRound1.context
 
   -- Input amount is exactly 100.0.
@@ -229,6 +231,7 @@
                       amount
                       lock = None
                       receiverFeeRatio = 1.0]
+        upgrade = Some ()
       context = transferContext
 
 testSenderChangeAndOutput: Script ()
@@ -271,6 +274,7 @@
                     amount = outputAmount3
                     lock = None
                 ]
+      upgrade = Some ()
     context = context.context
   assertMsg "senderChangeAmount > 0"
     (transferResult.summary.senderChangeAmount > 0.0)
@@ -320,6 +324,7 @@
                     amount = outputAmount3
                     lock = None
                 ]
+      upgrade = Some ()
     context = context.context
   assertMsg "senderChangeAmount == 0"
     (transferResult.summary.senderChangeAmount == 0.0)
@@ -361,6 +366,7 @@
                     lock = None
 
                 ]
+      upgrade = Some ()
     context = context.context
   assertMsg "senderChangeAmount == 0"
     (transferResult.summary.senderChangeAmount == 0.0)
diff -ur -x '*~' -x target -x dar -x .daml daml/canton-coin-test/daml/CC/Scripts/TestCoinRulesTransferLimits.daml daml/canton-coin-test-upgrade/daml/CC/Scripts/TestCoinRulesTransferLimits.daml
--- daml/canton-coin-test/daml/CC/Scripts/TestCoinRulesTransferLimits.daml
+++ daml/canton-coin-test-upgrade/daml/CC/Scripts/TestCoinRulesTransferLimits.daml
@@ -44,6 +44,7 @@
           provider = alice.primaryParty
           inputs = inputCoins
           outputs = []
+          upgrade = Some ()
         context = context
         expectedError = "More than the maximum number of inputs"
     return ()
@@ -63,6 +64,7 @@
           provider = alice.primaryParty
           inputs = [InputCoin coin]
           outputs = outputCoins
+          upgrade = Some ()
         context = context
         expectedError = "More than the maximum number of outputs"
     return ()
@@ -94,6 +96,7 @@
                 expiresAt = now `addRelTime` (minutes 1)
               receiverFeeRatio = 0.0
           ]
+        upgrade = Some ()
       context = context
       expectedError = "Fewer than the allowed number of lock holders"
     return ()
diff -ur -x '*~' -x target -x dar -x .daml daml/canton-coin-test/daml/CC/Scripts/TestComputeFees.daml daml/canton-coin-test-upgrade/daml/CC/Scripts/TestComputeFees.daml
--- daml/canton-coin-test/daml/CC/Scripts/TestComputeFees.daml
+++ daml/canton-coin-test-upgrade/daml/CC/Scripts/TestComputeFees.daml
@@ -124,6 +124,7 @@
                       amount = 10.0
                       lock = None
                       receiverFeeRatio = 1.0]
+        upgrade = Some ()
       context = context.context
 
   -- Actual holding fee should be based on the coin price when the coin was created, not the current one
diff -ur -x '*~' -x target -x dar -x .daml daml/canton-coin-test/daml/CC/Scripts/TestDesignExample.daml daml/canton-coin-test-upgrade/daml/CC/Scripts/TestDesignExample.daml
--- daml/canton-coin-test/daml/CC/Scripts/TestDesignExample.daml
+++ daml/canton-coin-test-upgrade/daml/CC/Scripts/TestDesignExample.daml
@@ -53,6 +53,7 @@
                      lock = None
                      receiverFeeRatio = 1.0
                  ]
+               upgrade = Some ()
          exerciseAppTransfer transferContext transfer
 
 -- | This matches the example in the design doc
@@ -187,10 +188,12 @@
       svc = app.svc
       owner = alice.primaryParty
       amount = expiringAmount transferConfig.holdingFee 1000.25 c2.round
+      upgrade = Some ()
   change === Coin with
     svc = app.svc
     owner = alice.primaryParty
     amount = expiringAmount transferConfig.holdingFee senderChangeAmount c2.round
+    upgrade = Some ()
 
   payment === Coin with
     svc = app.svc
@@ -198,6 +201,7 @@
     amount =
       let amount = arbiterPaymentAmount - transferredCoinFee
       in expiringAmount transferConfig.holdingFee amount c2.round
+    upgrade = Some ()
 
   -- check app reward issued by the escrow refresh
   [(_, appRewardCoupon)] <- queryFilter @AppRewardCoupon app.svc (\reward -> reward.round.number == 50002)
diff -ur -x '*~' -x target -x dar -x .daml daml/canton-coin-test/daml/CC/Scripts/TestLockAndCoinExpiry.daml daml/canton-coin-test-upgrade/daml/CC/Scripts/TestLockAndCoinExpiry.daml
--- daml/canton-coin-test/daml/CC/Scripts/TestLockAndCoinExpiry.daml
+++ daml/canton-coin-test-upgrade/daml/CC/Scripts/TestLockAndCoinExpiry.daml
@@ -255,6 +255,7 @@
                 expiresAt = now `addRelTime` lockDuration
               receiverFeeRatio = 0.0
           ]
+        upgrade = Some ()
   (openRound, _) <- getLatestOpenRound app
   featuredAppRight <- getFeaturedAppRight app owner.primaryParty
 
diff -ur -x '*~' -x target -x dar -x .daml daml/canton-coin-test/daml/CC/Scripts/TestRewardComputation.daml daml/canton-coin-test-upgrade/daml/CC/Scripts/TestRewardComputation.daml
--- daml/canton-coin-test/daml/CC/Scripts/TestRewardComputation.daml
+++ daml/canton-coin-test-upgrade/daml/CC/Scripts/TestRewardComputation.daml
@@ -88,6 +88,7 @@
       provider = provider1.primaryParty
       inputs = [ InputSvRewardCoupon svRewardBob2 ]
       outputs = [ ]
+      upgrade = Some ()
     context = context.context
 
 
@@ -99,6 +100,7 @@
       provider = bob.primaryParty
       inputs = [ InputSvRewardCoupon svRewardBob2 ]
       outputs = [ ]
+      upgrade = Some ()
     context = context.context
   transferResult.summary.senderChangeAmount ===
     ( intToDecimal bob2Weight * issuingMiningRound0.issuancePerSvRewardCoupon - createFee)
@@ -123,6 +125,7 @@
                     amount = 8.0
                     lock = None
                 ]
+      upgrade = Some ()
     context = context.context
 
   let expectedValidatorFaucetAmount = getIssuingMiningRoundIssuancePerValidatorFaucetCoupon issuingMiningRound0
diff -ur -x '*~' -x target -x dar -x .daml daml/canton-coin-test/daml/CC/Scripts/Util.daml daml/canton-coin-test-upgrade/daml/CC/Scripts/Util.daml
--- daml/canton-coin-test/daml/CC/Scripts/Util.daml
+++ daml/canton-coin-test-upgrade/daml/CC/Scripts/Util.daml
@@ -65,6 +65,7 @@
   _ <- submit svc $ createCmd CoinRules with
     configSchedule = defaultCoinConfigSchedule
     isDevNet = True
+    upgrade = Some ()
     ..
 
   submitExerciseCoinRulesByKey app [svc] [] CoinRules_Bootstrap_Rounds with
@@ -336,6 +337,7 @@
               receiverFeeRatio = transferFeeRatio
               lock = None
           ]
+        upgrade = Some ()
   (openRound, _) <- getLatestOpenRound app
   featuredAppRight <- getFeaturedAppRight app sender.primaryParty
   void $ submitExerciseCoinRulesByKey app [sender.primaryParty, recipient] readAs
@@ -364,6 +366,7 @@
                 expiresAt = now `addRelTime` lockDuration
               receiverFeeRatio = 0.0
           ]
+        upgrade = Some ()
   (openRound, _) <- getLatestOpenRound app
   featuredAppRight <- getFeaturedAppRight app sender.primaryParty
   void $ submitExerciseCoinRulesByKey app [sender.primaryParty, lockHolder] readAs
@@ -429,6 +432,7 @@
               map (InputSvRewardCoupon . fst) svRewardCoupons ++
               coins
           outputs = []
+          upgrade = Some ()
     -- merge multiple coins or rewards and claim rewards even if there are no coins yet
     when (length transfer.inputs > 1 || (length transfer.inputs == 1 && null coins)) $ void $
       submitExerciseCoinRulesByKey app [user] readAs CoinRules_Transfer with
diff -ur -x '*~' -x target -x dar -x .daml daml/canton-coin-test/daml.yaml daml/canton-coin-test-upgrade/daml.yaml
--- daml/canton-coin-test/daml.yaml
+++ daml/canton-coin-test-upgrade/daml.yaml
@@ -6,14 +6,14 @@
 sdk-version: 3.0.0-snapshot.20240209.12740.0.v499cb7f6
 name: canton-coin-test
 source: daml
-version: 0.1.0
+version: 0.2.0
 dependencies:
   - daml-prim
   - daml-stdlib
   - daml3-script
 data-dependencies:
   - ../cn-util/.daml/dist/cn-util-0.1.0.dar
-  - ../canton-coin/.daml/dist/canton-coin-0.1.0.dar
+  - ../canton-coin-upgrade/.daml/dist/canton-coin-0.2.0.dar
 build-options:
   - --ghc-option=-Wunused-binds
   - --ghc-option=-Wunused-matches
