diff -ur -x '*~' -x target -x dar -x .daml daml/wallet-payments/daml/CN/Wallet/Payment.daml daml/wallet-payments-upgrade/daml/CN/Wallet/Payment.daml
--- daml/wallet-payments/daml/CN/Wallet/Payment.daml
+++ daml/wallet-payments-upgrade/daml/CN/Wallet/Payment.daml
@@ -116,6 +116,7 @@
                       expiresAt = expiresAt
                     receiverFeeRatio = 0.0
                 ]
+              upgrade = Some ()
          result <- exercisePaymentTransfer context transfer
          let [TransferResultLockedCoin lockedCoin] = result.createdCoins
          let round = result.round
@@ -186,6 +187,7 @@
               provider
               inputs = [InputCoin ccsum.coin]
               outputs = map mkReceiverOutput ccReceiverAmounts
+              upgrade = Some ()
         result <- exerciseAppTransfer context transfer
         require "Precomputed fees were exact" (result.summary.senderChangeFee == 0.0)
         let coins = map (\(TransferResultCoin coinCid) -> coinCid) result.createdCoins
diff -ur -x '*~' -x target -x dar -x .daml daml/wallet-payments/daml/CN/Wallet/Subscriptions.daml daml/wallet-payments-upgrade/daml/CN/Wallet/Subscriptions.daml
--- daml/wallet-payments/daml/CN/Wallet/Subscriptions.daml
+++ daml/wallet-payments-upgrade/daml/CN/Wallet/Subscriptions.daml
@@ -324,6 +324,7 @@
                expiresAt = expiresAt
              receiverFeeRatio = 0.0
          ]
+       upgrade = Some ()
 
   result <- exercisePaymentTransfer context transfer
   let [TransferResultLockedCoin lockedCoin] = result.createdCoins
@@ -342,6 +343,7 @@
         provider
         inputs = [InputCoin ccsum.coin]
         outputs = [mkTransferOutput receiver targetAmount]
+        upgrade = Some ()
   result <- exerciseAppTransfer transferContext transfer
   require "Precomputed fees were exact" (result.summary.senderChangeFee == 0.0)
   let [TransferResultCoin coinCid] = result.createdCoins
diff -ur -x '*~' -x target -x dar -x .daml daml/wallet-payments/daml.yaml daml/wallet-payments-upgrade/daml.yaml
--- daml/wallet-payments/daml.yaml
+++ daml/wallet-payments-upgrade/daml.yaml
@@ -1,13 +1,13 @@
 sdk-version: 3.0.0-snapshot.20240209.12740.0.v499cb7f6
 name: wallet-payments
 source: daml
-version: 0.1.0
+version: 0.2.0
 dependencies:
   - daml-prim
   - daml-stdlib
 data-dependencies:
 - ../cn-util/.daml/dist/cn-util-0.1.0.dar
-- ../canton-coin/.daml/dist/canton-coin-0.1.0.dar
+- ../canton-coin-upgrade/.daml/dist/canton-coin-0.2.0.dar
 build-options:
   - --ghc-option=-Wunused-binds
   - --ghc-option=-Wunused-matches
