diff -ur -x '*~' -x target -x dar -x .daml ../daml/wallet/daml/CN/Wallet/Install.daml ../daml/wallet-upgrade/daml/CN/Wallet/Install.daml
--- ../daml/wallet/daml/CN/Wallet/Install.daml
+++ ../daml/wallet-upgrade/daml/CN/Wallet/Install.daml
@@ -140,6 +140,7 @@
         provider = endUserP
         inputs = inputs
         outputs = []
+        upgrade = Some ()
   transfer
 
 -- Enum of the possible ways coins can be manipulated through the wallet.
diff -ur -x '*~' -x target -x dar -x .daml ../daml/wallet/daml/CN/Wallet/TransferOffer.daml ../daml/wallet-upgrade/daml/CN/Wallet/TransferOffer.daml
--- ../daml/wallet/daml/CN/Wallet/TransferOffer.daml
+++ ../daml/wallet-upgrade/daml/CN/Wallet/TransferOffer.daml
@@ -99,6 +99,7 @@
                   lock = None
                   receiverFeeRatio = 0.0
               ]
+            upgrade = Some ()
         result <- exercisePaymentTransfer transferContext transfer
         archive self
         pure ((result, TransferOfferTrackingInfo { .. }), result.senderChangeCoin)
diff -ur -x '*~' -x target -x dar -x .daml ../daml/wallet/daml.yaml ../daml/wallet-upgrade/daml.yaml
--- ../daml/wallet/daml.yaml
+++ ../daml/wallet-upgrade/daml.yaml
@@ -1,15 +1,15 @@
 sdk-version: 2.9.0-snapshot.20231127.12418.0.v59ebefde
 name: wallet
 source: daml
-version: 0.1.0
+version: 0.2.0
 dependencies:
   - daml-prim
   - daml-stdlib
   - daml-script
 data-dependencies:
-- ../canton-coin/.daml/dist/canton-coin-0.1.0.dar
+- ../canton-coin-upgrade/.daml/dist/canton-coin-0.2.0.dar
 - ../cn-util/.daml/dist/cn-util-0.1.0.dar
-- ../wallet-payments/.daml/dist/wallet-payments-0.1.0.dar
+- ../wallet-payments-upgrade/.daml/dist/wallet-payments-0.2.0.dar
 build-options:
   - --ghc-option=-Wunused-binds
   - --ghc-option=-Wunused-matches
